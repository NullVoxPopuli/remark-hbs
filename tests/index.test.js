const { stripIndent } = require('common-tags');
const { assertTransform } = require('./utils');

test('can render text in a block component', () => {
  assertTransform({
    input: stripIndent`
      <Nested::Component>
        some text
      </Nested::Component>
    `,
  });
});

test('Nested components with nested components', () => {
  assertTransform({
    input: stripIndent`
      <Component::With::Nesting>
        <Nested::Component @foo="foo" @bar="bar" />
        <Nested::Component @foo="foo" @bar="bar" />
      </Component::With::Nesting>
    `,
  });
});

test('Nested component invocation within a named block with an empty body', () => {
  assertTransform({
    input: stripIndent`
      <Table>
        <:body as |Entry|>
          <Entry>
            Text
          </Entry>
        </:body>
      </Table>
    `,
  });
});

test('inline named blocks', () => {
  assertTransform({
    input: stripIndent`
      <Table::Nested>
        <:head>a</:head>
        <:body>b</:body>
        <:foot>
          foo
        </:foot>
      </Table::Nested>
    `,
  });
});

test('Nested component invocation with a named block / slot', () => {
  assertTransform({
    input: stripIndent`
      <Styleguide::DefinitionsTable>
        <:body as |Entry|>
          <Entry>
            text
          </Entry>
        </:body>
      </Styleguide::DefinitionsTable>
    `,
  });
});

test('Yielded components with html block content', () => {
  assertTransform({
    input: stripIndent`
      <Styleguide::Columns as |c|>
        <c.error>long string of text text<br/>text</c.error>
        <c.error>text<br/>text</c.error>
      </Styleguide::Columns>
    `,
  });
})

test('Yielded components with empty line within block', () => {
  assertTransform({
    input: stripIndent`
      <Styleguide::Columns as |c|>
        <c.error>
          long string of text text

          text
        </c.error>
      </Styleguide::Columns>
    `,
  });
})
