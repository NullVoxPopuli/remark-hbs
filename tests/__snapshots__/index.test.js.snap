// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it should keep all the HBS and HTML 1`] = `
"<h1>Test</h1>
<p>This is <em>testing</em> several features of ember templates</p>
<h2>Just html</h2>
<div>Something</div>
<h2>Simple angle brackets components</h2>
<MyComponent>
This is my component
</MyComponent>
<h2>Complex components</h2>
<MyForm as |form|>
<form.Input />
<form.Button type=\\"submit\\">Submit</form.Button>

</MyForm>
<OtherComponent @arg={{this.testing}} />
<ComponentWithHtml>
  <div class=\\"testing\\">testing</div>
</ComponentWithHtml>
<h2>Curry components</h2>
{{my-component isTesting=true}}
{{#other-component as |test|}}
{{test}}
{{/other-component}}
<h2>Curry values &#x26; helpers</h2>
{{this.myValue}}
{{value}}
{{array isTesting=true isCool=true}}
{{#each this.items as |item|}}
{{item}}
{{/each}}
{{#if this.isConditional}}
Yep
{{else}}
Nope
{{/if}}
<h2>Modifiers</h2>
<MyComponent {{did-insert this.coolStuff}} />
<div {{did-insert this.coolStuff}}>
  testing
</div>
<h2>Multi-line components</h2>
<FormRadioGroup
@label=\\"Grouped radio buttons\\"
@errors={{array \\"Nope\\"}}
as |Radio|>
<Radio
@value={{true}}
@label=\\"Yes\\" />
<Radio
@value={{false}}
@label=\\"No\\" />
<Radio
@value=\\"maybe\\"
@label=\\"Maybe\\" />

</FormRadioGroup>
<h2>Inline Code &#x26; Code Block with HBS</h2>
<p><code>\\\\{{this.something}</code></p>
<pre><code class=\\"language-hbs\\">&#x3C;FormRadioGroup
  @label=\\"Grouped radio buttons\\"
  @errors=\\\\{{array \\"1\\"}}
  as |Radio|
>
  &#x3C;Radio @value=\\\\{{true}} @label=\\"Yes\\" />
  &#x3C;Radio @value=\\\\{{false}} @label=\\"No\\" />
  &#x3C;Radio @value=\\"maybe\\" @label=\\"Maybe\\" />
&#x3C;/FormRadioGroup>
</code></pre>
"
`;
